From 100a5a07690623295d5b1e7fbd2af77523268cbe Mon Sep 17 00:00:00 2001
From: Steve Grubb <ausearch.1@gmail.com>
Date: Sat, 5 Sep 2026 15:26:08 -0400
Subject: [PATCH] Fix missing include of stddef.h

Indirectly import the glibc macros so that libc implementations that do not provide the macros use the fallbacks.
---
 utils/gcc-attributes.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/utils/gcc-attributes.h b/utils/gcc-attributes.h
index fa002930..23ce5370 100644
--- a/utils/gcc-attributes.h
+++ b/utils/gcc-attributes.h
@@ -4,7 +4,12 @@
 #ifndef GCC_ATTRIBUTES_H
 #define GCC_ATTRIBUTES_H
 
-#include <sys/cdefs.h>
+// Indirectly load libc attribute definitions through a portable public
+// header. Use the fallbacks below when the libc does not provide them.
+// The macros provide extra hints for gcc-analyzer and some compiler
+// optimizations. You're not missing any runtime security protections if
+// libc does not provide the macros.
+#include <string.h>
 
 /* These macros originate in sys/cdefs.h. Stub them when unavailable. */
 #ifndef __returns_nonnull
